forge/db/controllers/ProjectTemplate.js

```diff
@@ -0,0 +1,40 @@
+module.exports = {
+    validateSettings: function (app, settings) {
+        if (settings.httpAdminRoot !== undefined) {
+            let httpAdminRoot = settings.httpAdminRoot
+            delete settings.httpAdminRoot
+            if (typeof httpAdminRoot === 'string') {
+                httpAdminRoot = httpAdminRoot.trim()
+                if (httpAdminRoot.length > 0) {
+                    if (httpAdminRoot[0] !== '/') {
+                        httpAdminRoot = `/${httpAdminRoot}`
+                    }
+                    if (httpAdminRoot[httpAdminRoot.length - 1] === '/') {
+                        httpAdminRoot = httpAdminRoot.substring(httpAdminRoot.length - 2)
```

Comment:
Is this right?
```shell
$ node
Welcome to Node.js v16.14.0.
Type ".help" for more information.
> let httpAdmin = "/foo/"
undefined
> httpAdmin.substring(httpAdmin.length -2)
'o/'
> 
```
Should it be `httpAdminRoot = httpAdminRoot.substring(0, httpAdminRoot.length - 1)`

Reply:
```suggestion
                        httpAdminRoot = httpAdminRoot.substring(0, httpAdminRoot.length - 1)
```

Hmm, not sure what I was thinking there...